### PR TITLE
[NUI] Support Ellipsis Position Property 

### PR DESCRIPTION
**Pull request by shrouqsabah**
### Description of Change ###

Bind EllipsisPosition Property with DALi. 

**Patches:** 

- Support Ellipsis Position Property: https://review.tizen.org/gerrit/c/platform/core/uifw/dali-toolkit/+/259560 
- [NUI] Support Ellipsis Position Property: https://review.tizen.org/gerrit/c/platform/core/uifw/dali-csharp-binder/+/260614

Added new property Ellipsis to TextEditor
Added new property Ellipsis_Position to TextLabel, TextField and TextEditor

Handled Ellipsis location for below cases:

- Locations: END, START and MIDDLE
- TextController: TextLabel, TextField and TextEditor
- SingleLine and Muilt-lines with different LineWrap modes
- Text direction LTR and RTL

### API Changes ###

 - ACR: None

**Added:**
 - EllipsisPosition TextLabel.EllipsisPosition { get; set; } // Property
 - EllipsisPosition TextField.EllipsisPosition { get; set; } // Property
 - EllipsisPosition TextEditor.EllipsisPosition { get; set; } // Property
 - bool TextEditor.Ellipsis { get; set; } // Property

### CODE REVIEW

**Comment by wonrst** [comment]
Below patch should be applied first.
https://review.tizen.org/gerrit/c/platform/core/uifw/dali-toolkit/+/259560
https://review.tizen.org/gerrit/c/platform/core/uifw/dali-csharp-binder/+/260614

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 1]
```diff
@@ -1269,6 +1269,47 @@ public Tizen.NUI.PropertyMap Placeholder
             }
         }
 
+        /// <summary>
+        /// The Ellipsis property.<br />
+        /// Enable or disable the ellipsis.<br />
+        /// </summary>
+        /// <since_tizen> 6.5 </since_tizen>
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1276]: Please remove this.
You need to add this tag when your API is open to the public.
And the API level of tizen_6.5 is 9.
<since_tizen> 9 </since_tizen>

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 2]
```diff
@@ -1269,6 +1269,47 @@ public Tizen.NUI.PropertyMap Placeholder
...
+                NotifyPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// The ellipsis location of the text.
+        /// The ellipsis location type when the text size over the layout size.<br />
+        /// The ellipsis location: End, Start or Middle.<br />
+        /// </summary>
+        /// <since_tizen> 6.5 </since_tizen>
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1297]: Please remove this.

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs [hunk 3]
```diff
@@ -867,6 +867,31 @@ public enum VerticalLineAlignment
         Bottom
     }
 
+    /// <summary>
+    /// An enum of ellipsis location.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/Common/NUIConstants.cs:873]: Please add this. it should be hidden before the API is opened.
[EditorBrowsable(EditorBrowsableState.Never)]

Note: https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.editorbrowsableattribute?view=net-5.0

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs [hunk 4]
```diff
@@ -867,6 +867,31 @@ public enum VerticalLineAlignment
...
+    public enum EllipsisLocation
+    {
+        /// <summary>
+        /// ellipsis location at end.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/Common/NUIConstants.cs:879]: Please add this. [EditorBrowsable(EditorBrowsableState.Never)]

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs [hunk 5]
```diff
@@ -867,6 +867,31 @@ public enum VerticalLineAlignment
...
+        End,
+
+        /// <summary>
+        /// ellipsis location at start.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/Common/NUIConstants.cs:885]: Same as above.

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs [hunk 6]
```diff
@@ -867,6 +867,31 @@ public enum VerticalLineAlignment
...
+        Start,
+
+        /// <summary>
+        /// ellipsis location in middle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/Common/NUIConstants.cs:891]: Same as above.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs (hunk 1 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs]: Done.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs (hunk 2 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs]: Done

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs (hunk 3 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/Common/NUIConstants.cs]: Done

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs (hunk 4 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/Common/NUIConstants.cs]: Done

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs (hunk 5 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/Common/NUIConstants.cs]: Done

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs (hunk 6 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/Common/NUIConstants.cs]: Done

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 7]
```diff
@@ -1269,6 +1269,36 @@ public Tizen.NUI.PropertyMap Placeholder
             }
         }
 
+        /// This will be released at Tizen.NET API Level 9, so currently this would be used as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1273]: Current Master branch is for API Level 9, 
So, It need a description and need to remove EditorBrowsable

File: src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs [hunk 8]
```diff
@@ -781,13 +781,7 @@ public bool Ellipsis
...
         }
 
-        /// <summary>
-        /// The ellipsis location of the text.
-        /// The ellipsis location type when the text size over the layout size.<br />
-        /// The ellipsis location: End, Start or Middle.<br />
-        /// </summary>
-        /// <since_tizen> 6.5 </since_tizen>
-        /// This will be released at Tizen.NET API Level 6.5, so currently this would be used as inhouse API.
+        /// This will be released at Tizen.NET API Level 9, so currently this would be used as inhouse API.
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs:784]: Please add summary for EllipsisPosition.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 9]
```diff
@@ -1269,12 +1269,7 @@ public Tizen.NUI.PropertyMap Placeholder
             }
         }
 
-        /// <summary>
-        /// The Ellipsis property.<br />
-        /// Enable or disable the ellipsis.<br />
-        /// </summary>
-        /// <since_tizen> 6.5 </since_tizen>
-        /// This will be released at Tizen.NET API Level 6.5, so currently this would be used as inhouse API.
+        /// This will be released at Tizen.NET API Level 9, so currently this would be used as inhouse API.
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1272]: Please add summary.
You just need to remove the since_tizen tag.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 10]
```diff
@@ -1289,23 +1284,17 @@ public bool Ellipsis
...
         }
 
-        /// <summary>
-        /// The ellipsis location of the text.
-        /// The ellipsis location type when the text size over the layout size.<br />
-        /// The ellipsis location: End, Start or Middle.<br />
-        /// </summary>
-        /// <since_tizen> 6.5 </since_tizen>
-        /// This will be released at Tizen.NET API Level 6.5, so currently this would be used as inhouse API.
+        /// This will be released at Tizen.NET API Level 9, so currently this would be used as inhouse API.
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1287]: Please add summary.
You just need to remove the since_tizen tag.

File: src/Tizen.NUI/src/public/BaseComponents/TextField.cs [hunk 11]
```diff
@@ -1348,6 +1348,21 @@ public Color GrabHandleColor
             }
         }
 
+        /// This will be released at Tizen.NET API Level 9, so currently this would be used as inhouse API.
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/BaseComponents/TextField.cs:1351]: Please add summary.

Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextField.cs]: done

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs (hunk 10 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs]: done

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs (hunk 9 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs]: done

File: src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs (hunk 8 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs]: done

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs [hunk 12]
```diff
@@ -867,28 +867,11 @@ public enum VerticalLineAlignment
         Bottom
     }
 
-    /// <summary>
```
[the remark this answers is not in the document]
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/Common/NUIConstants.cs:870]: done

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs [hunk 13]
```diff
@@ -867,28 +867,11 @@ public enum VerticalLineAlignment
...
-    /// <since_tizen> 6 </since_tizen>
-    public enum EllipsisLocation
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public enum EllipsisPosition
     {
-        /// <summary>
-        /// ellipsis location at end.
-        /// </summary>
-        /// <since_tizen> 6 </since_tizen>
         End,
```
[the remark this answers is not in the document]
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/Common/NUIConstants.cs:880]: done

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs (hunk 7 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs]: I removed comment of API Level 9.

**Review by wonrst — APPROVED** [review]
Looks good to me.
This patch should be merged with csharp-binder.

**Review by wonrst — APPROVED** [review]

**Review by JoogabYun — APPROVED** [review]

**Review by huiyueun — APPROVED** [review]

**Pull request merged**